Lib/getpass.py

```diff
@@ -93,10 +108,13 @@ def unix_getpass(prompt='Password: ', stream=None):
         return passwd
 
 
-def win_getpass(prompt='Password: ', stream=None):
+def win_getpass(prompt='Password: ', stream=None, *, echochar=None):
     """Prompt for password with echo off, using Windows getwch()."""
     if sys.stdin is not sys.__stdin__:
         return fallback_getpass(prompt, stream)
+    if echochar and not echochar.isascii():
+        return ValueError(f"Invalid echochar: {echochar}. "
+                          "ASCII character expected.")
```

Comment:
Maybe let's have a single function checking validity of `echochar`.

Reply:
I like it, but I'm still not sure about the condition because '\b' is also ASCII character

Reply:
You can also exclude some control characters by checking the range then.